src/main/webapp/app/exercises/programming/manage/instructions-editor/analysis/programming-exercise-instruction-analysis.service.ts

```diff
@@ -113,19 +107,19 @@ export class ProgrammingExerciseInstructionAnalysisService {
      * @param analysis arbitrary number of analysis objects to be merged into one.
      */
     private mergeAnalysis = (...analysis: Array<AnalysisItem[]>) => {
-        return compose(
-            reduce((acc, [lineNumber, values, issueType]) => {
-                const lineNumberValues = acc[lineNumber];
-                const issueValues = lineNumberValues ? lineNumberValues[issueType] || [] : [];
-                return { ...acc, [lineNumber]: { ...lineNumberValues, [issueType]: [...issueValues, ...values] } };
-            }, {}),
-            map(([lineNumber, values, issueType]: AnalysisItem) => [
+        const reducer = (acc: ProblemStatementAnalysis, [lineNumber, values, issueType]: AnalysisItem) => {
```

Comment:
Why not using the existing `reduce()` from `lodash-es`?

Reply:
Same idea as mentioned with compose/flowRight as in the other comment.